platforms/enterprise/enterprise-operations/src/main/java/org/gradle/operations/dependencies/transforms/SkippedTransformExecutionProgressDetails.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.operations.dependencies.transforms;
+
+import org.gradle.operations.execution.ExecuteWorkBuildOperationType;
+
+/**
+ * Fired each time a transform execution request is obtained directly from the identity cache.
+ * <p>
+ * The transform execution will not happen at a later time.
+ *
+ * @since 8.7
+ */
+public interface SkippedTransformExecutionProgressDetails {
```

Comment:
Ah, I just read the Javadoc. If this is just about the identity cache, then I think that's what the name should indicate. `Skipped` tells me this can be skipped for a number of reasons.

I'd say, the purpose of the identity cache is to _deduplicate_ transform executions. We'll have a similar thing for non-incremental tasks at some point, too. Maybe that's what the name should suggest, e.g. `DeduplcatedTransformExecutionProgressDetails`?

Reply:
This is especially about transforms that not executed in the current build but taken from the identity cache from the previous build. We call those "SKIPPED" in build scans, that is why I used the same name here.
This is not about de-duplication, since this is for re-use between builds via the identity cache.
How about `IdentityCachedTransformExecutionProgressDetails`?

Reply:
Hm... That seems oddly specific. Why do we need to specifically care about transforms that are both a) present in the identity cache and b) are reused? It seems that we might be exposing a lot of internal behavior, no?

Reply:
The goal here is not to expose that some transforms did or did not execute in the current build. The goal is to expose origin build information for transforms that can't expose the information in the regular way, aka via an `ExecuteWorkBuildOperationType.Result`. For those transforms I need at least one of the above project events.
I don't even care too much when we evade the execution engine's "core", I only care that we have the information.

If we should expose the same information multiple times via this progress event then it doesn't matter. Or when we expose the information multiple times since we have an execution in this build and a progress event because of deduplication it doesn't matter, either.

> Why do we need to specifically care about transforms that are both a) present in the identity cache and b) are reused from the previous build?

We care about getting the origin information for transforms that circumvent the execution operation because the come from the identity cache. We don't have another way to obtain that information.